### PR TITLE
Implement phone based 2FA

### PR DESCRIPTION
**Pull request by Hattshire**

### CODE REVIEW

File: main/main.js [hunk 1]
```diff
@@ -48,20 +48,20 @@ function createWindow () {
   mainWindow.on('closed', () => mainWindow = null)
 }
 
-function createCheckpointWindow() {
-  const checkpointWindow = new BrowserWindow({
+function createPopupWindow(type) {
```
Comment (ifedapoolarewaju) [inline main/main.js]: can we change all occurrences of `popup` to `otp`? So we have `createOtpWindow`, `otp.js`, `otpWindow`, etc.

File: main/main.js [hunk 2]
```diff
@@ -103,16 +103,37 @@ function handleCheckpoint (checkpointError) {
...
   })
 }
 
+function handleTwoFactor (error) {
+    return new Promise((resolve, reject) => {
+        const tfWindow = createPopupWindow('twofactor')
+        const username = error.json.two_factor_info.username
+        const twoFactorIdentifier = error.json.two_factor_info.two_factor_identifier
+        const trustThisDevice = '1'
+        const verificationMethod = '1'
```
[the remark this answers is not in the document]
Comment (Hattshire) [reply main/main.js]: _trustThisDevice_ is for skipping the 2fa on the device next time user logs in(saving the device as trusted), a checkbox in the 2FA dialog is needed

_verificationMethod_ is unknown to me rn, I think it's the email verification vs phone/app verification, however it's '1' everytime ig logs me using 2fa in the app, the variable name is due to the key in the JSON. Gonna document it when I know it's meaning.
The upstream instagram-private-api has the 2FA implemented so I think will start there.

File: main/main.js (hunk 1 above)
Comment (Hattshire) [reply main/main.js]: I think so, there's no other login popup atm on the project

File: main/main.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hattshire) [reply main/main.js]: Here is a link to the 2FA doc on instagram-private-api's documentation:
https://github.com/dilame/instagram-private-api/blob/2b4d44167a3c9748dd88c95c0958ed2696067d23/docs/interfaces/_types_account_two_factor_login_options_.accounttwofactorloginoptions.md
Apparently they don't know either what other value would mean.

**Comment by ifedapoolarewaju** [comment]
thank you!

**Pull request merged**